### DISCUSSION TITLE
Custom Theme Support

### DISCUSSION THREAD

**Discussion by siamekanto19** [tailwindlabs/tailwindcss#8304] [Ideas]
It would be great if we have Custom Themes support in Tailwind. When we have dark theme configured in our config, we can generate dark mode classes with `dark` prefix (Ex: `dark:bg-gray-800`). It would be really great if we could generate classes for custom themes like that as well. 

As example, if we have`space` and `water` theme configured in our config file, we could generate classes for those themes using theme name prefix (`space:bg-gray-900 water:bg-blue-300`)  etc. 

If Tailwind can implement this, it would be revolutionary.

**Comment by aliharriss** [comment]
[This plugin](https://github.com/estevanmaito/tailwindcss-multi-theme#tailwind-css-multi-theme), using the [class strategy](https://tailwindcss.com/docs/dark-mode#toggling-dark-mode-manually) seems to allow more 'theme-variants' to be used in the way you want. It would be great to have this within tailwind though, instead of needing another dependency.
